libs/user_extensions/src/TTAlpsHistogramFiller.cpp

```diff
@@ -137,8 +94,15 @@ void TTAlpsHistogramFiller::FillDefaultVariables(const shared_ptr<Event> event)
 /// --------- NormCheck Histogram --------- ///
 
 void TTAlpsHistogramFiller::FillNormCheck(const shared_ptr<Event> event) {
-  float weight = nanoEventProcessor->GetGenWeight(asNanoEvent(event));
-  histogramsHandler->Fill("Event_normCheck", 0.5, weight);
+  histogramsHandler->Fill("Event_normCheck", 0.5);
+}
+
+/// --------- NormCheck Histogram --------- ///
+
+void TTAlpsHistogramFiller::FillDataCheck(const shared_ptr<Event> event) {
+  int isData = 0;
+  if (nanoEventProcessor->IsDataEvent(asNanoEvent(event))) isData = 1;
+  histogramsHandler->Fill("Event_isData", isData);
```

Comment:
why not this instead of the 3 lines above?

```cpp
histogramsHandler->Fill("Event_isData", nanoEventProcessor->IsDataEvent(asNanoEvent(event)));
```